packages/hierarchies-react/src/presentation-hierarchies-react/internal/TreeActions.ts

```diff
@@ -67,35 +115,22 @@ export class TreeActions {
   private loadSubTree(options: LoadNodesOptions, initialRootNode?: TreeModelRootNode) {
     const loadAction = this.getLoadAction(options.parent.id);
     const timeTracker = new TimeTracker((time) => this._onLoad(loadAction, time));
-    const parentId = options.parent.id;
-    this._loader
-      .loadNodes(options)
-      .pipe(collectTreePartsUntil(this._reset, initialRootNode))
-      .subscribe({
-        next: (newModel) => {
-          const childNodes = newModel.parentChildMap.get(parentId);
-          const firstChildNode = childNodes?.length ? newModel.idToNode.get(childNodes[0]) : undefined;
-          this.handleLoadedHierarchy(parentId, newModel);
-          // only report load duration if no error occurs
-          if (!(firstChildNode && isTreeModelInfoNode(firstChildNode))) {
-            timeTracker.finish();
-          }
-        },
-        complete: () => {
-          this.onLoadingComplete(parentId);
-          timeTracker[Symbol.dispose]();
-        },
-      });
+
+    return {
+      complete: new Promise<void>((resolve) => {
+        this._nodeLoader.next({ loadOptions: options, onComplete: resolve, timeTracker, parentId: options.parent.id, initialRootNode });
```

Comment:
the promise should be resolved and timetracker should be disposed for every request, including the ones that are debounced - does that happen?

Reply:
I think you could use `bufferTime` instead of `debounce`. That way you'd get all the requests, which would allow you to:
- merge their props (e.g. `state: "discard"` and `state: "keep"`),
- process the dropped ones.

Reply:
> bufferTime constantly emits values

Oh wow, never noticed that. I guess we could filter those out, but would be emitting those empty arrays would be a waste.

Reply:
Fixed. Used a combination of buffer and debounceTime.